File: TortillaRecetaBuilder.java

public class TortillaRecetaBuilder extends RecetaBuilder{
    public TortillaRecetaBuilder() {
        super.receta = new Receta();
    }

    @Override
    public void BuildIngredientes() {
        receta.setIngredientes("Huevos, patatas, cebolla, sal, aceite");
    }

    @Override
    public void BuildPasos() {
        receta.setPasos("Pelar y cortar las patatas y la cebolla. Batir los huevos y mezclar con las patatas y la cebolla. Freír en una sartén con aceite caliente.");
    }

    @Override
    public void BuildTiempo() {
        receta.setTiempoPreparacion(30);
       }
}

File: practica3/lib/tortillarecetabuilder.dart
import 'recetabuilder.dart';
import 'receta.dart';

class TortillaRecetaBuilder extends RecetaBuilder {
  TortillaRecetaBuilder() {
    super.receta = Receta('Tortilla de patatas');
  }

  @override
  void buildIngredientes() {
    receta.setIngredientes("Huevos, patatas, cebolla, sal, aceite");
  }

  @override
  void buildPasos() {
    receta.setPasos("Pelar y cortar las patatas y la cebolla. Batir los huevos y mezclar con las patatas y la cebolla. Freír en una sartén con aceite caliente.");
  }

  @override
  void buildTiempo() {
    receta.setTiempoPreparacion(30);
  }
}
